tendermint/src/amino_types/vote.rs

```diff
@@ -58,9 +58,9 @@ impl From<&vote::Vote> for Vote {
             vote_type: vote.vote_type.to_u32(),
             height: vote.height.value() as i64, // TODO potential overflow :-/
             round: vote.round as i64,
-            block_id: Some(BlockId {
-                hash: vote.block_id.hash.as_bytes().to_vec(),
-                parts_header: vote.block_id.parts.as_ref().map(PartsSetHeader::from),
+            block_id: vote.block_id.as_ref().map(|block_id| BlockId {
```

Comment:
As far as I know, the rust version of amino proto is like this:
* `None` is skiped, as the same as default value of protobuf.
* `Some(value)` encodes the value as is, won't skip it if it's the default value.
And the go version seems to automatically skip default values, so if rust version encodes `Some(default_value)`, it won't match the go version, we need to prevent this from happening.